### PR TITLE
CI Infra Docs

### PR DESCRIPTION
**Pull request by quasiben**
PR adds Infrastructure documentation including info on gpuCI .  

This PR may be reformatted as there is an ongoing effort to reorganize the Docs in https://github.com/dask/dask/pull/7984

cc @jrbourbeau

### CODE REVIEW

File: docs/source/develop.rst [hunk 1]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
...
+.  Note, the ``dask-build-environment`` has two separate Dockerfiles for Dask
+and Distributed similiarlly, gpuCI will run for both `Dask
+https://gpuci.gpuopenanalytics.com/job/dask/job/dask/job/prb/job/dask-prb/>`_
+and `Distributed
+<https://gpuci.gpuopenanalytics.com/job/dask/job/distributed/job/prb/job/distributed-prb/>`_
+
+For each PR, gpuCI will run all tests decorated with the pytest marker
+`@pytest.mark.gpu`.  This is configured in the `gpuci folder
+<https://github.com/dask/dask/tree/main/continuous_integration/gpuci>`_ .
+Like Github Actions, gpuCI will not run when first time authors to Dask or
```
Comment (jrbourbeau) [inline docs/source/develop.rst:328]:
```suggestion
Like Github Actions, gpuCI will not run when first time contributors to Dask or
```

File: docs/source/develop.rst [hunk 2]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
...
+Distributed submit PRs.  In this case, a gpuCI bot will comment on the PR:
```
Comment (jrbourbeau) [inline docs/source/develop.rst:329]:
```suggestion
Distributed submit PRs.  In this case, the gpuCI bot will comment on the PR:
```

File: docs/source/develop.rst [hunk 3]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
...
+
+> Can one of the admins verify this patch?
+
+Dask Maintainers can approved build these PRs with following choices:
```
Comment (jrbourbeau) [inline docs/source/develop.rst:333]:
```suggestion
Dask Maintainers can then approve gpuCI builds for these PRs with following choices:
```

File: docs/source/develop.rst [hunk 4]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
 ``git commit --no-verify`` or with the short version ``git commit -n``.
 
 
+Dask CI Infrastructure
```
Comment (jrbourbeau) [inline docs/source/develop.rst:291]: Can we move this section to the end of this page? Currently it's between the "Contributing to Code" and "Contributing to Documentation" sections, which seems a little out of place

File: docs/source/develop.rst [hunk 5]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
...
+Distributed submit PRs.  In this case, a gpuCI bot will comment on the PR:
+
+> Can one of the admins verify this patch?
+
+Dask Maintainers can approved build these PRs with following choices:
+
+- To approve PR author for just the current PR, comment the command:
+  - ``ok to test``
+- To approve PR author for all PRs in the future, comment the command:
+  - ``add to allowlist``
```
Comment (jrbourbeau) [inline docs/source/develop.rst:338]: This seems slightly more clear to me, though feel free to push back

```suggestion
- To only approve the PR contributor for the current PR, leave a comment which states ``ok to test``
- To approve the current PR and all future PRs from the contributor, leave a comment which states ``add to allowlist``
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks for adding this @quasiben!

**Comment by quasiben** [comment]
Thanks for the feedback @jrbourbeau -- I'll update with another commit shortly addressing the rest of your thoughts

File: docs/source/develop.rst [hunk 6]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
...
+Dask CI Infrastructure
+----------------------
+
+Github Actions
+~~~~~~~~~~~~~~
+
+Dask uses Github Actions for Continuous Integration (CI) testing for each PR.
+The actions will run the test suite across a variety of Python versions, Operating
+Systems, and NumPy and Pandas versions.  Additionally, Dask can upstream Xarray tests
+with the commit message ``test-upstream``
```
[the remark this answers is not in the document]
Comment (quasiben) [reply docs/source/develop.rst:300]: +1

File: docs/source/develop.rst [hunk 7]
```diff
@@ -288,6 +288,58 @@ will be run each time you commit changes. You can skip these checks with
...
+
+
+GPU CI
+~~~~~~
+
+As Dask seamlessly work with GPUs, Dask will also test PRs with a GPU enabled
+CI environment provided by NVIDIA: `gpuCI <https://gpuci.gpuopenanalytics.com/>`_ .
+Unlike Github Actions, the CI environment for gpuCI is controlled with a `docker image
+<https://github.com/rapidsai/dask-build-environment/>`_ .  When making commits
+to the repo, a new image is built.  The docker image building process can be
```
[the remark this answers is not in the document]
Comment (quasiben) [reply docs/source/develop.rst]: Updated in ba5527359

**Comment by quasiben** [comment]
@jrbourbeau do you have preferences in showing the gputester message ?

Screen Shot 2021-08-04 at 11 22 11 AM

**Comment by charlesbluca** [comment]
Said this [here](https://github.com/dask/community/issues/175#issuecomment-892954038), but it might be nice to have a condensed version of this (particularly the comment system) in a PR template on both repos; okay with handling that in follow up PRs depending on what direction we go with gpuCI

**Comment by jrbourbeau** [comment]
If you don't mind, I'd prefer to hold off on merging this until https://github.com/dask/distributed/issues/5154 is resolved

> @jrbourbeau do you have preferences in showing the gputester message ?

Neat, I personally like the image version. Do you have a preference? 

> Said this here, but it might be nice to have a condensed version of this info (particularly how the comments work) in a PR template on both repos;

I'd prefer to point to our developer guide instead. That will minimize the number of places we need to update when something related to our development process changes.

**Comment by quasiben** [comment]
> If you don't mind, I'd prefer to hold off on merging this until dask/distributed#5154 is resolved

Yup, no rush.   I think I like the image better

**Comment by GenevieveBuckley** [comment]
I have a preference, can the bot message be: "Can one of the admins verify this patch for GPU testing?"

If we don't include "for GPU testing" then new contributors are likely to still get confused. I don't think it's enough to expect people to look at the bot name and put two and two together, that's too subtle.

Thanks for doing all of this @quasiben

File: docs/source/develop.rst [hunk 8]
```diff
@@ -330,4 +330,60 @@ The resulting HTML files end up in the ``build/html`` directory.
...
+``@pytest.mark.gpu``.  This is configured in the `gpuci folder
+<https://github.com/dask/dask/tree/main/continuous_integration/gpuci>`_ .
+Like Github Actions, gpuCI will not run when first time contributors to Dask or
+Distributed submit PRs.  In this case, the gpuCI bot will comment on the PR:
+
+.. note:: Can one of the admins verify this patch?
+
+.. image:: images/gputester-msg.png
+
+Dask Maintainers can then approve gpuCI builds for these PRs with following choices:
```
Comment (fjetter) [inline docs/source/develop.rst:380]: What kind of github permission level does this require? Afaik, there are `Read`, `Triage`, `Write`, `Maintain` and `Admin` privileges, we might not be using all of them. From this passage, I would guess it is `Maintain`? see https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization

Related question: What is the default permission for dask org `member`s?

**Comment by quasiben** [comment]
> I have a preference, can the bot message be: "Can one of the admins verify this patch for GPU testing?"

 I'm +1 on this change as well and I'll ask OPs about our options

> Thanks for doing all of this @quasiben

Thanks should go to @charlesbluca and @raydouglass for connecting all of this up

File: docs/source/develop.rst (hunk 8 above)
Comment (quasiben) [reply docs/source/develop.rst:380]: Currently, this is limited to folks who [Dask owners](https://github.com/orgs/dask/people?query=role%3Aowner) but we don't have to be limited to that group.  gpuCI can be manually manipulated to include additional folks who may not be a Dask owner but still need/require/want this capability.  

I think default permissions for dask members is the same as public permissions: read/create PSs/issues/comments/etc but no write access

**Comment by jakirkham** [comment]
Was there anything else needed here?

**Comment by charlesbluca** [comment]
> Was there anything else needed here?

@jrbourbeau suggested we wait for a resolution on dask/distributed#5154 before merging this; that being said, I'm not sure how long the current solution (ops changing the bot message) will take, so I'm in favor of merging this now and then updating the docs later on as gpuCI changes.

**Review by jrbourbeau — APPROVED** [review]
Thanks for the ping @jakirkham. I agree with @charlesbluca, let's merge this in, but keep https://github.com/dask/distributed/issues/5154 open to track updating the bot messages. When that happens we can update the corresponding parts here. 

Thanks @quasiben for writing this up!

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**